DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -311,30 +311,71 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Channel.Parse(response, category);
     }
 
-    public async IAsyncEnumerable<ChannelThread> GetChannelThreadsAsync(
-        Snowflake channelId,
+    public async IAsyncEnumerable<ChannelThread> GetGuildThreadsAsync(
+        Snowflake guildId,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        var currentOffset = 0;
-        while (true)
+        var channels = (await GetGuildChannelsAsync(guildId, cancellationToken))
+            .ToArray();
+        var tokenKind = _resolvedTokenKind ??= await GetTokenKindAsync(cancellationToken);
+        
+        if (tokenKind == TokenKind.Bot)
         {
-            var url = new UrlBuilder()
-                .SetPath($"channels/{channelId}/threads/search")
-                .SetQueryParameter("offset", currentOffset.ToString())
-                .Build();
-
-            var response = await TryGetJsonResponseAsync(url, cancellationToken);
-            if (response is null)
-                break;
-
-            foreach (var threadJson in response.Value.GetProperty("threads").EnumerateArray())
+            var response = await TryGetJsonResponseAsync($"guilds/{guildId}/threads/active", cancellationToken);
```

Comment:
Can this actually return a non-200 response? Or would it just be an empty array in the response?

If the latter, then we should just use `GetJsonResponseAsync(...)` instead.

Reply:
Seems it's the latter, yeah. I tested a few different scenarios, and it doesn't seem like it'll ever return a non-200, except in cases where the guild is inaccessible by the bot. Resolved in d586bfd.